src/Components/Modals/DeleteActivationKeyConfirmationModal.js

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import {
+  Button,
+  Modal,
+  ModalVariant,
+  TextContent,
+  Text,
+  TextVariants,
+} from '@patternfly/react-core';
+import { ExclamationTriangleIcon } from '@patternfly/react-icons';
+import propTypes from 'prop-types';
+import useDeleteActivationKey from '../../hooks/useDeleteActivationKey';
+import useNotifications from '../../hooks/useNotifications';
+import Loading from '../LoadingState/Loading';
+import { useQueryClient } from 'react-query';
+
+const DeleteActivationKeyConfirmationModal = (props) => {
+  const { isOpen, handleModalToggle, name } = props;
+  const { addSuccessNotification, addErrorNotification } = useNotifications();
+  const { mutate, isLoading } = useDeleteActivationKey();
+  const queryClient = useQueryClient();
+
+  const deleteActivationKey = (name) => {
+    mutate(name, {
+      onSuccess: (data, name) => {
+        queryClient.resetQueries('activation_keys');
```

Comment:
What do you think about just removing the the deleted key from the table instead of reloading the whole table? The benefit of reloading the table is that we're guaranteed to get the most up to date list instead of just what is currently in the table.

I'll leave it up to you.

```javascript
      onSuccess: (_data, name) => {
        queryClient.setQueryData('activation_keys', (oldData) =>
          oldData.filter((entry) => entry.name != name)
        );
```